internal/controller/controllers/events_uploader_handler.go

```diff
@@ -0,0 +1,304 @@
+package controllers
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"net/textproto"
+	"strings"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/openshift/assisted-service/internal/common"
+	eventsapi "github.com/openshift/assisted-service/internal/events/api"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/k8sclient"
+	"github.com/sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type serializedAuthMap struct {
+	Auths map[string]serializedAuth `json:"auths"`
+}
+type serializedAuth struct {
+	Auth string `json:"auth"`
+}
+
+type EventsUploaderConfig struct {
+	DataUploadEndpoint     string `envconfig:"DATA_UPLOAD_ENDPOINT" default:""`
+	AssistedServiceVersion string
+	EnableDataCollection   bool `envconfig:"ENABLE_DATA_COLLECTION" default:"true"`
```

Comment:
why true by default?

Reply:
It was decided that this would be enabled by default unless the user explicitly opts out to mimic the behavior of telemetry (decision #3 in [kick off meeting doc](https://docs.google.com/document/d/1AAkKfr4acWVuk6GyPL901dSbd1XKaFaFA7vSQQZkyn0/edit?usp=sharing))

Do you think it should actually be disabled by default instead?